controller.rst

```diff
@@ -587,6 +587,94 @@ using the ``type`` option of the attribute::
 
     The ``type`` option of ``#[MapRequestPayload]`` was introduced in Symfony 7.1.
 
+Mapping Uploaded File
+~~~~~~~~~~~~~~~~~~~~~
+
+You can map ``UploadedFile``s to the controller arguments and optionally bind ``Constraints`` to them.
+The argument resolver fetches the ``UploadedFile`` based on the argument name.
+
+.. code-block:: php-attributes
+
+    namespace App\Controller;
+
+    use Symfony\Component\HttpFoundation\File\UploadedFile;
+    use Symfony\Component\HttpFoundation\Response;
+    use Symfony\Component\HttpKernel\Attribute\MapUploadedFile;
+    use Symfony\Component\Routing\Attribute\Route;
+    use Symfony\Component\Validator\Constraints as Assert;
+
+    #[Route('/user/picture', methods: ['PUT'])]
+    class UserPictureController
+    {
+        public function _invoke(
+            #[MapUploadedFile(new Assert\File(mimeTypes: ['image/png', 'image/jpeg']))]
+            UploadedFile $picture
+        ): Response {
+            // ...
+        }
+    }
+
+.. tip::
+
+    The bound ``Constraints`` are performed before injecting the ``UploadedFile`` into the controller argument.
+    When a constraint violation is detected an ``HTTPException`` is thrown and the controller's
+    action is not executed.
+
+Mapping ``UploadedFile``s with no custom settings.
+
+.. code-block:: php-attributes
```

Comment:
Are these directives really needed, as there is only one code block and not tabs?

Reply:
I had to explicitly set `php-attributes` markup format to these code blocks to pass the checkers.